src/main/java/data/Deadline.java

```diff
@@ -16,6 +16,17 @@ public Deadline(String name, String by) {
         this.byDate = by;
     }
 
+    /**
+     * Getter for by date of Deadline task.
+     * Overrides super dummy function.
+     *
+     * @return The String representation of the by date.
+     */
+    @Override
+    public String getByDate() {
```

Comment:
You could consider transitioning to LocalDate or LocalDateTime type to make the dates more robust.